NewForce-Cohort-6/tabloidfullstack-cranberry-sauce
# Jg list posts by category 43

# Description
[Ticket number 43](https://trello.com/c/jRNy3yQt/43-43-list-posts-by-category) added the  functionality for viewing posts by category. For the time being I’ve put this in category management and added a button next to each category in the list that takes you to the list of posts in that category. 



# Testing Instructions

1.  `git add` and `git commit` whatever you're working on on your branch
2.  `git checkout main`
3. ` git fetch --a`
4.  `git checkout jg-ListPostsByCategory-43` to switch to my branch.
5. If you don't see the most recent version of the files, try` git pull origin jg-ListPostsByCategory-43`.
6. If you haven't created the Tabloid FullStack database, please do so.
7.  Use this to fill out tables and columns: [SQL data](https://github.com/NewForce-Cohort-6/tabloidfullstack-cranberry-sauce/blob/main/SQL/01_Tabloid_Create_DB.sql)
8.  Use this to seed the database: [SQL Seed Data](https://github.com/NewForce-Cohort-6/tabloidfullstack-cranberry-sauce/blob/main/SQL/02_Tabloid_Seed_Data.sql)
9. Execute Tabloid from inside of Visual Studio by clicking on the solid green arrow. This will launch Swagger and run the back end of the application to provide the API that will be utilized to view the front end. 
10. From your terminal run npm start from the `client` directory within `tabloidfullstack-cranberry-sauce`
11.  Step 10 will bring up the front end of the application in a browser window titled React App. In the url of that browser page, please Login using an email that you know is there and works for you (example: foo@bar.com from seed data)
12. Navigate to the Category Management page using the nav bar.  You should see a list of Category names printed down the page. With a ‘See Posts’ button next to each.
13. The categories “Cooking & Baking”, “Music”, “Politics”, “Science”, “Tech” should all have posts associated with them from the seed data. The others do not. Choose one of these posts to click the button next to.  
14. You should be brought to a page that lists the posts titles (nonsense titles from seed data). Each title has a button next to it that, in the future, will show the post’s details. However now it navigates to a page that says “Not yet implemented.” It will be implemented with ticket 5. 

Confirmation of step 14 means the feature works. 

# Checklist:
